src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -123,21 +123,26 @@ class SourcesTree extends Component<Props, State> {
     }
 
     if (nextProps.shownSource && nextProps.shownSource != shownSource) {
-      const listItems = getDirectories(nextProps.shownSource, sourceTree);
+      const matchingSources = Object.keys(sources).filter(
+        sourceId => getRawSourceURL(sources[sourceId].url) === shownSource
+      );
 
-      if (listItems && listItems[0]) {
+      if (matchingSources.length) {
+        const listItems = getDirectories(
+          sources[matchingSources[0]],
```

Comment:
lets see if we can use `getSource` here and we might need to return if there isn't a source...